core/src/main/java/io/jenkins/pluginhealth/scoring/probes/IncrementalBuildDetectionProbe.java

```diff
@@ -0,0 +1,134 @@
+
+                if (firstExtensionPath.isPresent()) {
+                    return isExtensionsXMLConfigured(firstExtensionPath.get())
+                        ? ProbeResult.success(key(), "Incremental Build is configured in the plugin.")
+                        : ProbeResult.failure(key(), "Incremental Build is not configured in the plugin.");
+                } else if (firstConfigPath.isPresent()) {
+                    return isMavenConfigConfigured(firstConfigPath.get())
+                        ? ProbeResult.success(key(), "Incremental Build is configured in the plugin.")
+                        : ProbeResult.failure(key(), "Incremental Build is not configured in the plugin.");
+                } else {
+                    return ProbeResult.failure(key(), "Incremental Build is not configured in the plugin.");
+                }
+            } catch (IOException ex) {
+                LOGGER.error("Could not browse the plugin folder during probe {}. {}", key(), ex);
+                return ProbeResult.error(key(), "Could not browse the plugin folder.");
+            }
+        } catch (IOException ex) {
+            LOGGER.error("Could not browse the plugin folder during probe {}. {}", key(), ex);
+            return ProbeResult.error(key(), "Could not browse the plugin folder.");
+        }
+    }
+
+    @Override
+    public String[] getProbeResultRequirement() {
+        return new String[]{ContinuousDeliveryProbe.KEY};
+    }
+
+    @Override
+    public String key() {
+        return KEY;
+    }
+
+    @Override
+    public String getDescription() {
+        return "The probe detects whether incremental build is configured in Maven Config.";
+    }
+
+    private boolean isExtensionsXMLConfigured(Path path) {
+        try {
+            DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
+            DocumentBuilder dBuilder = dbFactory.newDocumentBuilder();
+            Document doc = dBuilder.parse(path.toFile());
+            doc.getDocumentElement().normalize();
+            Element rootElement = doc.getDocumentElement();
+            Element extensionElement = (Element) rootElement.getElementsByTagName("extension").item(0);
+            Element groupIdElement = (Element) extensionElement.getElementsByTagName("groupId").item(0);
+            String groupId = groupIdElement.getTextContent();
+            return groupId.equals(INCREMENTAL_TOOL);
+        } catch (IOException e) {
+            LOGGER.error("Could not read the file during probe {}. {}", key(), e);
+        } catch (ParserConfigurationException | SAXException e) {
+            LOGGER.error("Could not parse the file during probe {}. {}", key(), e);
+        }
+        return false;
+    }
+
+    private boolean isMavenConfigConfigured(Path path) {
+        try {
+            return Files.lines(path).anyMatch(line -> line.contains("-Pconsume-incrementals") || line.contains("-Pmight-produce-incrementals"));
```

Comment:
This is not doing what we want. We want to check that both profiles are present in the file, not only one of them.

Reply:
Here changing `||` to `&&` wasn't working. I had to change more of the syntax.